src/Compilers/CSharp/Portable/CSharpParseOptions.cs

```diff
@@ -67,7 +67,7 @@ public override IEnumerable<string> PreprocessorSymbolNames
             this.SpecifiedLanguageVersion = languageVersion;
             this.LanguageVersion = languageVersion.MapSpecifiedToEffectiveVersion();
             this.PreprocessorSymbols = preprocessorSymbols.ToImmutableArrayOrEmpty();
-            _features = features?.ToImmutableDictionary() ?? ImmutableDictionary<string, string>.Empty;
+            _features = features?.ToImmutableDictionary(StringComparer.OrdinalIgnoreCase) ?? ImmutableDictionary<string, string>.Empty;
```

Comment:
Did we use case-insensitive comparison here before? It doesn't feel right to change case-sensitivity of the comparison. If it was culture specific, then we should simply switch to Ordinal or Invariant.